app/controllers/adg_registrations_controller.rb

```diff
@@ -19,6 +19,12 @@ def create
       adg_answer.save!
     end
     current_user.update_attribute(:adg_preference_ids, params[:adg_registration][:adg_preference_ids])
+
+    adg_template = CmsPage.where(reference_string: 'Email::Adg').first    
+    if adg_template
+      UserMailer.welcome_email(current_user, adg_template).deliver
+    end
+    
```

Comment:
I'd prefer to see this mailer logic tucked away somewhere like a UserNotifier class, or somesuch
```
...
UserNotifier.send_adg_email
UserNotifier.send_welcome_email
...

Prob not a great idea to hack the welcome_email method for the adg one. But if you need to do some refactoring to dry things up, by all means do.